### DISCUSSION TITLE
Teleport for ssh and https application behind NAT

### DISCUSSION THREAD

**Discussion by ghost** [gravitational/teleport#5775] [Q&A] [answered]
Hello,
I have about 200 linux machines in different locations. 95% of them behind NAT. I thought to create a cluster for these nodes with ssh and https app_service.

Question is how to configure it on proxy and node site? I have to make 200 apps with different names and 200 DNS CNAME's for every https app and on the nodes configure app for the location?
Then there will be for example:
https1.teleport.example.com
https2.teleport.example.com
https3.teleport.example.com
Etc..

Is that good or wrong?

Tkank You for your attention.

Regards,
Rafał

**Comment by klizhentas** [comment, marked as answer]
hey Rafał, good to meet you. You don't have to create 200 CNAME entries, have you considered `*.teleport.example.com` wildcard DNS entry? You can connect 200 nodes and register them within the same cluster.

**Reply by ghost** [reply]
Of course. Shame on me. I forgot about wildcard CNAME's. Thanks a lot.
I will try this tomorrow.

**Reply by klizhentas** [reply]
@mrkarpinski no problem at all, let me know how it goes :)
